### PR TITLE
Implementation of invite sending + lots of refactoring work

### PR DESCRIPTION
**Pull request by Fang-**
This is kind of all over the place. @jtobin some of this we already looked at closely. Would be great if you could give everything else a _quick_ look-over. It's mostly straightforward refactoring/cleanup work, but it's kind of all over the place...

### CODE REVIEW

**Comment by shrugs** [comment]
there's a merge conflict committed to the `mino` branch at the moment. if you nuke your local `node_modules` and re-run `npm install` it should iron itself out.

**Comment by shrugs** [comment]
I also get an issue around the `websocket` module that web3-providers depends on - after an `npm install --save websocket` the project is happy. Do you experience the same?

**Comment by Fang-** [comment]
I didn't understand what you meant, but did it anyway. npm picked it up for me: `A git conflict was detected in package-lock.json.` Fixed, and ran a quick `npm audit fix` while I was at it.

Haven't seen the `websocket` issue before...

**Comment by shrugs** [comment]
How do I make sure my planets have invites in order to test the sending flow a bit? I've got some planets on my testnet, but none of them have invites to give.

**Comment by Fang-** [comment]
You log in as a star and grant them invites (and make sure the star's spawn proxy is configured, the "manage invites" page warns you about this).

Give me a couple minutes and I'll get you a .patch that makes `InvitesSend.js` not depend on the email sender!

**Comment by Fang-** [comment]
[Use this!](https://gist.github.com/Fang-/ceacf8f0df94adc1635b704d51553daa) It just sends the generated ticket(s) to browser cli instead of doing a dance to send them by email.

**Comment by shrugs** [comment]
@Fang- nice! generates the invite ticket. I wasn't able to get the transaction itself to go through during "send invite" but I don't know nearly enough about the contracts themselves to effectively debug this.

**Comment by Fang-** [comment]
>I wasn't able to get the transaction itself to go through during "send invite" but I don't know nearly enough about the contracts themselves to effectively debug this.

That's very strange! The UI should catch all cases of not being able to send according to contract logic. Might want to make sure you have latest azimuth-solidity and azimuth-js in node_modules, that's bitten me like this before. If that's not it, let's talk tomorrow!

@jtobin thanks, I'll get to these first thing in the morning. (:

File: src/bridge/Bridge.js [hunk 1]
```diff
@@ -212,6 +217,11 @@ class Bridge extends React.Component {
   }
 
   setWallet(wallet) {
+    if (Just.hasInstance(wallet) &&
```
[the remark this answers is not in the document]
Comment (Fang-) [reply src/bridge/Bridge.js]: Didn't know about Maybe.map, thanks!

File: src/walletgen/lib/lib.js [hunk 2]
```diff
@@ -53,20 +53,28 @@ const generateWallet = async (point, ticket, boot) => {
...
-  // This is here to notify the anyone who opens console because the thread
-  // hangs, blocking UI updates so this cannot be doen in the UI
+  // This is here to notify anyone who opens console because the thread
+  // hangs, blocking UI updates so this cannot be done in the UI
   console.log('Generating Wallet for point address: ', point);
 
   return wallet;
 }
 
+const generateOwnershipWallet = async (point, ticket) => {
```
[the remark this answers is not in the document]
Comment (Fang-) [reply src/walletgen/lib/lib.js]: I kinda like the explicit `async` still, it keeps you from having to add a "returns a promise" comment. But eh.

File: src/bridge/lib/tank.js [hunk 3]
```diff
@@ -34,8 +38,84 @@ const fundTransactions = signedTxs => {
...
+// gotFunding: optional callback that stops telling the user to go get funding
+const ensureFundsFor = async (web3, point, address, cost, signedTxs,
+                              askForFunding, gotFunding) => {
+  let balance = await web3.eth.getBalance(address);
+
+  if (cost > balance) {
+
+    try {
+
+      if (point !== null) {
```
[the remark this answers is not in the document]
Comment (Fang-) [reply src/bridge/lib/tank.js]: Not per se. A todo comment further down explain how this part of the code might be safe to just tear out. I don't feel warmly enough about either `maybe` and this piece of code to change this right now.

**Comment by Fang-** [comment]
(Nearly) all done! I'll go ahead and merge this.

**Pull request merged**